### PR TITLE
Update mage cauterize and alter time, and some other CC abilities

### PR DESCRIPTION
**Pull request by SweepyBoop**
Alter time needs to use the spell id of the buffs, and cooldown starts when the buff fades

Cauterize needs to use the spell id of the debuff.

Tested both in skirmishes

### CODE REVIEW

**Comment by SweepyBoop** [comment]
Hi Vendethiel, I noticed 2 major mage abilities are incorrect and made a fix. Tested in skirmishes

**Review by vendethiel — APPROVED** [review]

**Comment by vendethiel** [comment]
Need to test for a bit myself, then I'll merge. Thanks.

**Comment by SweepyBoop** [comment]
Please don't merge yet, I noticed cauterize is not always working, sometimes it is not triggering.
Strange that it is working fine for party frame but sometimes not for arena frame.
Here is the party frame, cauterize is triggering correctly.
image

The time cauterize didn't work, the mage ice blocked exactly the same time as cauterize proc. That might be why?

**Comment by SweepyBoop** [comment]
Strange that it is working for party frames but not for arena frames..

**Comment by SweepyBoop** [comment]
image
Clip from rated arena last night, alter time is working for arena frames.

**Comment by SweepyBoop** [comment]
Ah.. both are working unreliably.. They are only triggering sometimes.
I got 2 questions for these spells:

1. Spells that are based on proc debuff, like cauterize, are there other examples that I can test? I suspect they might be in similar situation?
2. Spells that trigger cooldown when buff fades, do they work properly if the buff is purged, or goes away prematurely?

I'd like to test other spells that are similar to cauterize and alter time and see if they have similar issues.

**Comment by vendethiel** [comment]
You can alias a spell id to another, via `LCT_SpellData[alias] = original`, instead of assigning a lua table. Maybe it’s better to keep both IDs that way.

**Comment by SweepyBoop** [comment]
For the alter time, the reason it's only working sometimes might be (after checking the source code):
SPELL_AURA_REMOVED is Triggered When Buffs/Debuffs expire (according to documentation here: https://wowpedia.fandom.com/wiki/COMBAT_LOG_EVENT)

So it will only register if it expires, but will not trigger if it's purged.

**Comment by SweepyBoop** [comment]
I'm pretty sure the spell IDs in this PR are correct since they are triggering for party frames correctly.
Some logic in the Lib.xml might needs some minor tweak to make spells like these work.

**Comment by SweepyBoop** [comment]
> You can alias a spell id to another, via `LCT_SpellData[alias] = original`, instead of assigning a lua table. Maybe it’s better to keep both IDs that way.

How does this work for cauterize and alter time?
For cauterize, is it LCT_SpellData[Cauterize] = spell id of cauterize debuff, or the other way around?
For alter time, is it LCT_SpellData[alter time] = spell id of alter time buff, or the opposite?

I'm pretty sure the spell IDs in this PR are correct since they are triggering for party frames correctly.
Some logic in the Lib.xml might needs some minor tweak to make spells like these work.
Meanwhile I'll also try using alias and see if it works reliably for these 2 abilities.

**Comment by SweepyBoop** [comment]
image
Alter time seems working now with latest iteration. The enemy mage used it 3 times and it triggered every time.

**Comment by SweepyBoop** [comment]
Latest iteration seems working for both abilities so far. I'll keep testing.

**Comment by SweepyBoop** [comment]
Working on updating more CC abilities, but will publish in next PR.
Everything in this PR (latest iteration) has been working so far, appreciate if you can test a bit too before merging.

**Comment by SweepyBoop** [comment]
Sadly still can't make cauterize and alter time work 100% reliably.
Reverting changes for these 2 spells and keeping the rest which are working properly.

I'll try to figure out cauterize and alter time in next PR.

**Comment by vendethiel** [comment]
Thanks!

**Pull request merged**
